docs/reactivity/reactive-components.md

```diff
@@ -0,0 +1,92 @@
+# Reactive components
+
+
+Components are organized by domains, for example components in a product domain, and can leverage domain logic to communicate with the associated backend API. Each domain is shipped with a domain service that provides an API to communicate with a backend API. For example, when rendering product data, `ProductService` can be used as follows:
+
+```ts
+export class ProductPriceComponent extends LitElement {
+  protected productService = resolve(ProductService);
+  protected product$ = this.productService.get({ sku: '123' });
+}
+```
+
+To ensure that components are reusable in different contexts, it is recommended to not couple them directly with the qualifier that is used to load data. For example, `ProductPriceComponent`does not need to be aware of the `SKU`, as the `SKU` can be determined from the route on the product page, the product card in a list, or the cart entry. Oryx provides a mechanism to setup a so-called context. In the case of product components, the product controller is used to resolve the SKU from the context controller.
```

Comment:
What do you mean by the `route`?

Reply:
We can change it to URL actually. In frontend development we often speak about _routing_, which is the mechanism to reflect the application state by the URL. The route however does not include the domain name, protocol, etc. which is why we typically do not talk about the URL.